File: backend/internal/services/buyer_service.go
package services

import (
	"context"

	"github.com/grawizah/backend/internal/interfaces"
	"github.com/grawizah/backend/internal/models"
)

type BuyerService struct {
	buyerRepo interfaces.BuyerRepository
}

func NewBuyerService(buyerRepo interfaces.BuyerRepository) *BuyerService {
	return &BuyerService{
		buyerRepo: buyerRepo,
	}
}

func (s *BuyerService) GetBuyerRadar(ctx context.Context, limit, offset int) ([]models.Buyer, error) {
	return s.buyerRepo.GetAll(limit, offset)
}

func (s *BuyerService) GetBuyerByID(ctx context.Context, id string) (*models.Buyer, error) {
	return s.buyerRepo.GetByID(id)
}

func (s *BuyerService) SearchBuyers(ctx context.Context, query string) ([]models.Buyer, error) {
	return s.buyerRepo.Search(models.BuyerSearchCriteria{
		Query: query,
	})
}

func (s *BuyerService) GetBuyersByCountry(ctx context.Context, country string) ([]models.Buyer, error) {
	return s.buyerRepo.GetByCountry(country)
}

func (s *BuyerService) GetHighQualityBuyers(ctx context.Context, limit int) ([]models.Buyer, error) {
	return s.buyerRepo.GetHighQualityBuyers(limit)
}

func (s *BuyerService) CalculateLeadScore(ctx context.Context, buyerID string, productCategory string) (map[string]interface{}, error) {
	buyer, err := s.buyerRepo.GetByID(buyerID)
	if err != nil {
		return nil, err
	}

	// Simple lead scoring algorithm
	buyScore := buyer.BuyScore

	return map[string]interface{}{
		"buyer_id":               buyerID,
		"conversion_probability": float64(buyScore) / 100.0,
		"buy_score":              buyScore,
		"factors": map[string]interface{}{
			"import_frequency": 0.85,
			"volume_trend":     0.78,
			"category_match":   0.90,
			"country_affinity": 0.75,
		},
	}, nil
}

File: frontend/src/services/BuyerService.ts
import { BaseService } from './BaseService';
import { Buyer, LeadScoringResult } from '@/types/buyer';

export class BuyerService extends BaseService {
  private readonly endpoint = '/api/buyers';

  async getBuyerRadar(filters?: {
    country?: string;
    min_score?: number;
    category?: string;
  }): Promise<Buyer[]> {
    return this.get<Buyer[]>(`${this.endpoint}/radar`, { params: filters });
  }

  async getBuyerById(id: string): Promise<Buyer> {
    return this.get<Buyer>(`${this.endpoint}/${id}`);
  }

  async searchBuyers(query: string): Promise<Buyer[]> {
    return this.post<Buyer[]>(`${this.endpoint}/search`, { query });
  }

  async getLeadScore(buyerId: string, productCategory: string): Promise<LeadScoringResult> {
    return this.post<LeadScoringResult>(`${this.endpoint}/${buyerId}/lead-score`, {
      product_category: productCategory
    });
  }

  async getBuyersByCountry(country: string): Promise<Buyer[]> {
    return this.get<Buyer[]>(`${this.endpoint}/country/${country}`);
  }

  async getTopBuyers(limit: number = 10): Promise<Buyer[]> {
    return this.get<Buyer[]>(`${this.endpoint}/top`, { params: { limit } });
  }
}
